### DISCUSSION TITLE
Pass optional parameter to a method conditionally

### DISCUSSION THREAD

**Discussion by kai-franke** [neuefische/questions#327] [Web] [answered]
Hello!

I use `splice()` to either delete an object from or add a new one to an array. 

`splice(start, deleteCount, item1)`
As described at [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice?retiredLocale=de), I 
- always pass the index as `start`,
- to **delete** `1` as `deleteCount`
- and to **add** a new item `0` as `deleteCount` and an object as `item`.

So I only want to pass the new item as third parameter to the method if I get a `deleteCount = 0`. Right now it looks like this in my function:
```
deleteCount
      ? data.splice(start, deleteCount)   // removes an item
      : data.splice(start, deleteCount, item1);   // adds an item
```

Is there a way to include the condition to the parameters? I thought of something like this:
```
data.splice(start, deleteCount, (!deleteCount && item1))
```

I would be very happy to learn if it is generally possible to perform calculations or conditions within the parameters and what the syntax would be for this.

**Comment by Kheirah** [comment]
`data.splice(start, deleteCount, !deleteCount ? item1 : undefined)`

It's not necessarily pleasant to the eye but it's a valid solution and exactly what you are asking for. You can use the ternary operator and pass on the `item` or an `undefined` value (which is equivalent to passing nothing).

**Reply by kai-franke** [reply]
Thanks a lot, @Kheirah! 
I have tried a few things based on your suggested solution, but unfortunately it doesn't work that way. Also an `undefined` is apparently "something", which in my case creates a new object when mapping the array, replacing the deleted one. The same is true for `null` or an empty string, by the way.
I have now decided to use an if-else statement.
Still, thanks for your input - it taught me a lot about parameters and arguments yesterday. 😃

**Reply by Kheirah** [reply]
Oh my bad, you're right, Kai! I hadn't tested `splice` and my reasoning was off.

Any value passed to the `splice` function starting from the third parameter is spreaded internally (which I wasn't aware of). Hence, anything entered is passed on regardless of its value.

Thank you Thomas for providing the answer 😄.

**Comment by ThomasF85** [comment, marked as answer]
Hi Kai,

I know of no pleasant way of writing your expression in a "one-liner".
however, it is best to only use the ternary operator, if you use its value. As you know the ternary operator evaluates to a value, that you can use lateron:
```
const value = deleteCount
      ? data.splice(start, deleteCount)
      : data.splice(start, deleteCount, item1)
```
If you do not use the value at all, it can be misleading to use the ternary operator here.
So if the only thing you want to do is to manipulate the array `data`, it would be better to use an if statement instead.

**Reply by kai-franke** [reply]
Thanks a lot, @ThomasF85!
Since `splice()` returns an array containing the deleted elements which I don't need as I'm only working with the manipulated array, I decided to use the if-else statement. 
I didn't know that the ternary operator should only be used when the return value is to be used. I thought it could also be a shorthand notation for an if-else statement. Thanks for the explanation! 😃
